### PR TITLE
Feature/bus 826 vimeo video cms

### PR DESCRIPTION
**Pull request by carojansing**
Adds Vimeo Cms Block

### CODE REVIEW

File: components/cms/block/CmsBlockVimeoVideo.vue [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<script setup lang="ts">
+import type { Schemas } from '@shopware/api-client/api-types';
+import CmsLoader from '~/components/cms/CmsLoader.vue';
```
Comment (aheartforspinach) [inline components/cms/block/CmsBlockVimeoVideo.vue:3]:
```suggestion
```

File: components/cms/element/CmsElementVimeoVideo.vue [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<script setup lang="ts">
+import type { CmsElementVimeoVideo } from '@shopware-pwa/composables-next';
+import Player from '@vimeo/player';
+const props = defineProps<{
+    element: CmsElementVimeoVideo;
+}>();
+const vimeoVideo = ref(null);
```
Comment (aheartforspinach) [inline components/cms/element/CmsElementVimeoVideo.vue:7]:
```suggestion
import Player from '@vimeo/player';

const props = defineProps<{
    element: CmsElementVimeoVideo;
}>();

const vimeoVideo = ref(null);
```

File: components/cms/element/CmsElementVimeoVideo.vue [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+onMounted(() => {
+    if (vimeoVideo.value) {
+        const player = new Player('vimeo-video', {
+            id: parseInt(props.element.config.videoID.value),
```
Comment (aheartforspinach) [inline components/cms/element/CmsElementVimeoVideo.vue:12]: could a value possible be null?

Comment (carojansing) [reply components/cms/element/CmsElementVimeoVideo.vue:14]: I believe it can't be null but my IDE marked it red.

Comment (mloeffle) [reply components/cms/element/CmsElementVimeoVideo.vue:14]: According to the typedefinition it should be null form my understanding, but i'd suggest doing a null check beforehand does not have a downside and we are on the safer side.

Comment (aheartforspinach) [reply components/cms/element/CmsElementVimeoVideo.vue:14]: and also use the formatting mentioning here https://github.com/basecom/Pond/pull/52#discussion_r1773364124

File: components/cms/element/CmsElementVimeoVideo.vue [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            width: vimeoVideo.value?.offsetWidth ?? null,
+        });
+        player.on('play', function () {
+        });
```
Comment (aheartforspinach) [inline components/cms/element/CmsElementVimeoVideo.vue:18]:
```suggestion
```

we can delete this or why do we need it?

**Review by mloeffle — APPROVED** [review]

**Pull request merged**